submissions/Vlada-khom/js-dom/main.js

```diff
@@ -0,0 +1,134 @@
+        with imported long-haired varieties, like the Turkish Angora and what today is called the 
+        Traditional Persian, with an aim to producing more stout and round-faced stock, while retaining 
+        the short coat. As a result of this hybridization, British catteries have frequently produced 
+        (generally unwanted) semi-long-haired offspring among their litters. In more recent years, 
+        these have been intentionally bred (often outside the UK) to each other and sometimes to 
+        standard British Shorthairs, to establish a consistent, formalized British Longhair breed.`,
+    ],
+    image: "./db/british-long.jpg",
+  },
+  {
+    name: "British Shorthair",
+    id: '5',
+    description: [
+      `The British Shorthair is the pedigreed version of the traditional British domestic 
+        cat, with a distinctively stocky body, dense coat, and broad face. The most familiar colour 
+        variant is the "British Blue", with a solid grey-blue coat, orange eyes, and a medium-sized 
+        tail. The breed has also been developed in a wide range of other colours and patterns, including 
+        tabby and colourpoint.`,
+      `It is one of the most ancient cat breeds known. In modern times, it remains the most popular 
+        pedigreed breed in its native country, as registered by the UK's Governing Council of the Cat 
+        Fancy (GCCF). A quarter of all kittens registered with the GCCF each year are British 
+        Shorthairs, making the British the most popular pedigree cat in the UK.`,
+      `The breed's good-natured appearance and relatively calm temperament make it a frequent media 
+        star, notably as the inspiration for John Tenniel's famous illustration of the Cheshire Cat 
+        from Alice in Wonderland. The Cat Fanciers' Association profile reads: "When gracelessness is 
+        observed, the British Shorthair is duly embarrassed, quickly recovering with a 'Cheshire cat 
+        smile'.`,
+    ],
+    image: "./db/british-short.jpg",
+  },
+];
+
+
+const content = document.querySelector(".content");
+
+function main() {
+  const sideWrap = document.querySelector(".sideWrap");
+  const catList = document.createElement("ul");
+  catList.className = "catList";
+  data.map((cat) => {
+    catList.innerHTML += `<li><div class="itemMenu" data-id="${cat.id}">${cat.name}</div></li>`;
+  }).join('');
+  sideWrap.appendChild(catList);
+
+  document.addEventListener('click', function (event) {
+    if(event.target.dataset.id) {
+      data.map((cat) => {
+        if (event.target.dataset.id === cat.id) {
+          content.innerHTML = `<h2 class="titleContent">${cat.name}</h2>
+                                <div class="boxImg"><img class="contentImg"src="${cat.image}" 
+                                alt="${cat.name}"></div> `;
+          cat.description.map((desc) => {
+            content.innerHTML += `<p>${desc}</p>`;
+          }).join(' ');
+        }
+      }).join('');
```

Comment:
If you just want to walk through the array and insert content, then you need to use the method `Array.forEach()`.
If you want to go through the array and find the desired element, wrap the result in the appropriate template, then you need to use the method `Array.map()` or `Array.reduce()`.
```javascript
const templateContent = data.map(cat => {
    if(cat.id == id) {
       return `<label>${cat.name}</label>`;
    }
}).join('');
content.innerHTML = templateContent;
```